### PR TITLE
fix: issue when GitHub organization contains more than 30 repos

### PR DESCRIPTION
**Pull request by KoblerS**
Fixed issue when GitHub organization contains more than 30 repositories. Scaler will now iterate over the pages according to GitHubs documentation.

### Checklist

- [x] When introducing a new scaler, I agree with the [scaling governance policy](https://github.com/kedacore/governance/blob/main/SCALERS.md)
- [x] I have verified that my change is according to the [deprecations & breaking changes policy](https://github.com/kedacore/governance/blob/main/DEPRECATIONS.md)
- [x] Tests have been added
- [x] Changelog has been updated and is aligned with our [changelog requirements](https://github.com/kedacore/keda/blob/main/CONTRIBUTING.md#Changelog)
- [x] A PR is opened to update our Helm chart ([repo](https://github.com/kedacore/charts)) *(if applicable, ie. when deployment manifests are modified)*
- [x] A PR is opened to update the documentation on ([repo](https://github.com/kedacore/keda-docs)) *(if applicable)*
- [x] Commits are signed with Developer Certificate of Origin (DCO - [learn more](https://github.com/kedacore/keda/blob/main/CONTRIBUTING.md#developer-certificate-of-origin-signing-your-work))

Fixes https://github.com/kedacore/keda/issues/5738

### CODE REVIEW

File: pkg/scalers/github_runner_scaler.go [hunk 1]
```diff
@@ -470,32 +470,44 @@ func (s *githubRunnerScaler) getRepositories(ctx context.Context) ([]string, err
...
+			url = fmt.Sprintf("%s/orgs/%s/repos?page=%s", s.metadata.githubAPIURL, s.metadata.owner, page)
+		case REPO:
+			url = fmt.Sprintf("%s/users/%s/repos?page=%s", s.metadata.githubAPIURL, s.metadata.owner, page)
+		case ENT:
+			url = fmt.Sprintf("%s/orgs/%s/repos?page=%s", s.metadata.githubAPIURL, s.metadata.owner, page)
+		default:
+			return nil, fmt.Errorf("runnerScope %s not supported", s.metadata.runnerScope)
+		}
+	
+		body, err := getGithubRequest(ctx, url, s.httpClient)
```
Comment (JorTurFer) [inline pkg/scalers/github_runner_scaler.go:489]: `getGithubRequest` returns 3 items
```suggestion
		body, _, err := getGithubRequest(ctx, url, s.httpClient)
```

**Review by JorTurFer — COMMENTED** [review]
Nice improvement! Could you include a unit test for this case too?

**Comment by KoblerS** [comment]
@JorTurFer sure will work on it!

**Review by zroubalik — COMMENTED** [review]
@KoblerS could you please fix the DCO problem?

**Comment by KoblerS** [comment]
> @KoblerS could you please fix the DCO problem?

could you please check again. Thanks!

File: CHANGELOG.md [hunk 2]
```diff
@@ -52,6 +52,8 @@ To learn more about active deprecations, we recommend checking [GitHub Discussio
 
 ## Unreleased
 
+- **GitHub Scaler**: Fixed pagination, fetching repository list ([#5738](https://github.com/kedacore/keda/issues/5738))
```
Comment (zroubalik) [inline CHANGELOG.md:55]: Please, move this to Improvements section below.

**Review by zroubalik — COMMENTED** [review]
There are also some problems in the imports:

```
@@ -2,13 +2,13 @@ package scalers
 
 import (
 	"context"
+	"crypto/rand"
 	"encoding/json"
 	"fmt"
-	"crypto/rand"
+	"html/template"
 	"math/big"
 	"net/http"
 	"net/http/httptest"
-	"html/template"
 	"strings"
 	"testing"
 	"time"
 	```

**Comment by KoblerS** [comment]
> There are also some problems in the imports:
> 
> ```
> @@ -2,13 +2,13 @@ package scalers
>  
>  import (
>  	"context"
> +	"crypto/rand"
>  	"encoding/json"
>  	"fmt"
> -	"crypto/rand"
> +	"html/template"
>  	"math/big"
>  	"net/http"
>  	"net/http/httptest"
> -	"html/template"
>  	"strings"
>  	"testing"
>  	"time"
> ```

what do you mean?

**Comment by zroubalik** [comment]
@KoblerS you can see the failed static check output: https://github.com/kedacore/keda/actions/runs/9127598269/job/25098204457?pr=5746

**Comment by KoblerS** [comment]
> @KoblerS you can see the failed static check output: https://github.com/kedacore/keda/actions/runs/9127598269/job/25098204457?pr=5746

sorry didn't saw it, should be fixed now.

**Comment by KoblerS** [comment]
> /run-e2e github **Update:** You can check the progress [here](https://github.com/kedacore/keda/actions/runs/9128477423)

Any updates when this will be merged?

**Review by zroubalik — APPROVED** [review]

**Pull request merged**